src/ring_theory/nilpotent.lean

```diff
@@ -44,6 +44,10 @@ end
 @[simp] lemma is_nilpotent_neg_iff [ring R] : is_nilpotent (-x) ↔ is_nilpotent x :=
 ⟨λ h, neg_neg x ▸ h.neg, λ h, h.neg⟩
 
+lemma is_nilpotent.map [monoid_with_zero R] [monoid_with_zero S] {r : R}
+  (hr : is_nilpotent r) (f : monoid_with_zero_hom R S) : is_nilpotent (f r) :=
+by { use hr.some, erw [← f.to_monoid_hom.map_pow, hr.some_spec, f.map_zero] }
```

Comment:
I agree with @jcommelin, let's make this generic and at the same time get rid of the `erw`. Unfortunately I hadn't made `map_pow` generic yet so you'll get a dependency on #10749:
```suggestion
lemma is_nilpotent.map [monoid_with_zero R] [monoid_with_zero S] {r : R}
  {F : Type*} [monoid_with_zero_hom_class F R S] (hr : is_nilpotent r) (f : F) :
  is_nilpotent (f r) :=
by { use hr.some, rw [← map_pow, hr.some_spec, map_zero] }
```

Reply:
Sure! I can wait.